src/datasources/tag/TagDataSource.ts

```diff
@@ -73,18 +78,61 @@ export class TagDataSource extends DataSourceBase<TagQuery> {
 
       return result
     } else {
-      const history = await this.getTagHistoryValues(tag.path, tag.workspace ?? tag.workspace_id, range, maxDataPoints);
-      result.fields = [
-        { name: 'time', values: history.datetimes },
-        { name, values: history.values },
-      ];
-    }
+      const workspaceTagMap: Record<string, TagWithValue[]> = {};
+      const tagPropertiesMap: Record<string, Record<string, string> | null> = {};
 
-    if (query.properties) {
-      result.fields = result.fields.concat(this.getPropertiesAsFields(tag.properties));
-    }
+      tagsWithValues.forEach(tagWithValue => {
+        let workspace = tagWithValue.tag.workspace ?? tagWithValue.tag.workspace_id
+        if (!workspaceTagMap[workspace]) {
+          workspaceTagMap[workspace] = [];
+        }
+        workspaceTagMap[workspace].push(tagWithValue);
+        tagPropertiesMap[`${getWorkspaceName(workspaces, workspace)}.${tagWithValue.tag.path}`] = tagWithValue.tag.properties
+      });
+      let tagsDecimatedHistory: { [key: string]: TypeAndValues } = {}
+      for (const workspace in workspaceTagMap) {
+        const tagHistoryResponse = await this.getTagHistoryWithChunks(
+          workspaceTagMap[workspace],
+          workspace,
+          range,
+          maxDataPoints
+        )
+        for (const path in tagHistoryResponse.results) {
+          // If tags are mixed from different workspaces, add the workspace name as a prefix
+          if (Object.keys(workspaceTagMap).length === 1) {
+            tagsDecimatedHistory[path] = tagHistoryResponse.results[path]
+          } else {
+            tagsDecimatedHistory[`${getWorkspaceName(await this.getWorkspaces(), workspace)}.${path}`] = tagHistoryResponse.results[path]
```

Comment:
I think we should only format it as `workspace.path` if there are duplicate paths. We should also consider doing this for the "Current" query to be consistent.

Reply:
@TigranVardanyan can you make this change?

Reply:
@markacamps done
@mure for current we can add additional column for workspace and when it will be necessary to combine with the name through the Grafana transformations

Reply:
@TigranVardanyan - If the change has been made, can this conversation be resolved?

Reply:
yes